avm/src/vm/memory.rs

```diff
@@ -136,15 +144,19 @@ impl HandlerMemory {
     event_id: i64,
     curr_addr: i64,
     curr_hand_mem: &Arc<HandlerMemory>,
-  ) -> Option<Arc<HandlerMemory>> {
-    let pls = Program::global().event_pls.get(&event_id).unwrap().clone();
+  ) -> VMResult<Option<Arc<HandlerMemory>>> {
```

Comment:
We're wrapping another layer around `HandlerMemory` here.

You know when we want to improve performance we're going to be undoing a lot of this and going back to "we have thoroughly tested/audited our code so we don't have to pay a runtime penalty for this safety" :/

Reply:
that makes sense, although many of these checks are happening anyways when we call `unwrap` or `expect`. This is just to allow us to prevent these panics and recover from them if the daemon is running.